File: src/app/interfaces/theme/LAFCommitter.java
package app.interfaces.theme;

import app.interfaces.dialog.ErrorMessage;

import java.awt.*;
import java.util.ArrayList;
import java.util.Arrays;

public class LAFCommitter {
  private static ArrayList<Frame> jfxs = new ArrayList<>();

  public LAFCommitter(Frame... jfxs) {
    LAFCommitter.jfxs.addAll(java.util.Arrays.asList(jfxs));
  }

  public void setMultTheme(Refresh theme) {
    try {
      for (Frame jfts : jfxs) {
        if (theme == null) {
          new ErrorMessage("Unable to process the specified Refresh theme.\nTrace Dump: "
              + Arrays.toString(jfxs.toArray()) + "\nArgument Type:" + jfts.getClass().getName());
          return;
        }
        theme.refresh(jfts);

      }
    } catch (Exception ex) {
      // do nothing
    }
  }
}

File: kt/app/interfaces/theme/LAFCommitter.kt
package app.interfaces.theme

import app.interfaces.dialog.ErrorMessage
import java.awt.Frame
import java.util.*

class LAFCommitter(vararg jfxs: Frame?) {
    fun setMultTheme(theme: Refresh?) {
        try {
            for (jfts in jfxs) {
                if (theme == null) {
                    ErrorMessage("""
    Unable to process the specified Refresh theme.
    Trace Dump: ${Arrays.toString(jfxs.toTypedArray())}
    Argument Type:${jfts.javaClass.name}
    """.trimIndent())
                    return
                }
                theme.refresh(jfts)
            }
        } catch (ex: Exception) {
            // do nothing
        }
    }

    companion object {
        private val jfxs = ArrayList<Frame>()
    }
    
}


